deepchem/models/keras_model.py

```diff
@@ -460,28 +476,49 @@ def fit_generator(self,
 
       if checkpoint_interval > 0 and current_step % checkpoint_interval == checkpoint_interval - 1:
         manager.save()
+        for ext_logger in self.loggers:
+          if isinstance(ext_logger, WandbLogger):
+            ext_logger.save_checkpoint(self.model_dir,
```

Comment:
It's up to the logger to decide when to write checkpoints.  `checkpoint_interval` tells the model when it should write checkpoints using the standard mechanism.  If a logger is writing checkpoints with a different mechanism, that isn't the model's concern.

Reply:
I was under the assumption that the Logger would use the same checkpoint_interval and strategy as the main model class. If that's not the case then we can have each logger will define its own checkpointing mechanism. Which means for WandbLogger we can initialize it with it's own checkpoint_interval argument. Is my understanding correct?

Reply:
Correct!